org.eclipse.wildwebdeveloper/src/org/eclipse/wildwebdeveloper/JSEditorAssociationOverride.java

```diff
@@ -0,0 +1,73 @@
+/*******************************************************************************
+ * Copyright (c) 2020 Red Hat Inc. and others.
+ *
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *   Victor Rubezhny (Red Hat Inc.) - initial implementation
+ *******************************************************************************/
+package org.eclipse.wildwebdeveloper;
+
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+
+import org.eclipse.core.runtime.content.IContentType;
+import org.eclipse.ui.IEditorDescriptor;
+import org.eclipse.ui.IEditorInput;
+import org.eclipse.ui.ide.IEditorAssociationOverride;
+import org.eclipse.ui.internal.genericeditor.ExtensionBasedTextEditor;
+import org.eclipse.ui.internal.genericeditor.GenericEditorWithContentTypeIcon;
+
+public class JSEditorAssociationOverride implements IEditorAssociationOverride {
+
+	private Map<String, IEditorDescriptor> descriptorMap = new HashMap<>();
+
+	@Override
+	public IEditorDescriptor[] overrideEditors(IEditorInput editorInput, IContentType contentType,
+			IEditorDescriptor[] editorDescriptors) {
+		return editorInput != null ? overrideEditors(editorInput.getName(), contentType, editorDescriptors)
+				: editorDescriptors;
+	}
+
+	@Override
+	public IEditorDescriptor[] overrideEditors(String fileName, IContentType contentType,
+			IEditorDescriptor[] editorDescriptors) {
+		return Arrays.stream(editorDescriptors).map(descriptor -> getEditorDescriptorForFile(descriptor, fileName)).toArray(size -> new IEditorDescriptor[size]);
+	}
+
+	@Override
+	public IEditorDescriptor overrideDefaultEditor(IEditorInput editorInput, IContentType contentType,
+			IEditorDescriptor editorDescriptor) {
+		return editorInput != null ? overrideDefaultEditor(editorInput.getName(), contentType, editorDescriptor)
+				: editorDescriptor;
+	}
+
+	@Override
+	public IEditorDescriptor overrideDefaultEditor(String fileName, IContentType contentType,
+			IEditorDescriptor editorDescriptor) {
+		return getEditorDescriptorForFile(editorDescriptor, fileName);
+	}
+
+	// Workaround for https://bugs.eclipse.org/bugs/show_bug.cgi?id=559907
+	private IEditorDescriptor getEditorDescriptorForFile(IEditorDescriptor defaultDescriptor, String fileName) {
+		if (defaultDescriptor != null && !ExtensionBasedTextEditor.GENERIC_EDITOR_ID.equals(defaultDescriptor.getId())
+				&& fileName != null && !fileName.isEmpty()) {
+			int index = fileName.lastIndexOf('.');
+			if (index > -1) {
+				String extension = fileName.substring(index);
+				if (".js".equalsIgnoreCase(extension.toLowerCase())) {
```

Comment:
We need to use ContentType rather than filenames. Content-Types have hierarchy, can be extended, customized by other plugins and users...
So we should just get the Wild Web Developer content type with `PlatformUI.getContentTypeManager().getContentType("org.eclipse.wildwebdeveloper.js")` then ideally compare it with the content-type given as method parameters, or -if it doesn't work- try `contentType.isAssociatedWith(fileName)`.

Reply:
I'll try to use this approach with `contentType.isAssociatedWith(fileName)`.